x-pack/plugins/cases/server/client/stats/client.ts

```diff
@@ -50,37 +49,31 @@ async function getStatusTotalsByType(
   params: CasesStatusRequest,
   clientArgs: CasesClientArgs
 ): Promise<CasesStatusResponse> {
-  const { unsecuredSavedObjectsClient, caseService, logger, authorization } = clientArgs;
+  const { caseService, logger, authorization } = clientArgs;
 
   try {
     const queryParams = pipe(
       excess(CasesStatusRequestRt).decode(params),
       fold(throwErrors(Boom.badRequest), identity)
     );
 
-    const { filter: authorizationFilter, ensureSavedObjectsAreAuthorized } =
-      await authorization.getAuthorizationFilter(Operations.getCaseStatuses);
+    const { filter: authorizationFilter } = await authorization.getAuthorizationFilter(
+      Operations.getCaseStatuses
+    );
+
+    const options = constructQueryOptions({
+      owner: queryParams.owner,
+      authorizationFilter,
+    });
 
-    // casesStatuses are bounded by us. No need to limit concurrent calls.
-    const [openCases, inProgressCases, closedCases] = await Promise.all([
-      ...caseStatuses.map((status) => {
-        const statusQuery = constructQueryOptions({
-          owner: queryParams.owner,
-          status,
-          authorizationFilter,
-        });
-        return caseService.findCaseStatusStats({
-          unsecuredSavedObjectsClient,
-          caseOptions: statusQuery,
-          ensureSavedObjectsAreAuthorized,
-        });
-      }),
-    ]);
+    const statusStats = await caseService.getCaseStatusStats({
```

Comment:
I think when we gonna provide KPIs for all cases this should be moved to the metrics classes.